src/App.jsx

```diff
@@ -57,102 +67,140 @@ export function App({ gmApiKey }) {
       setInitialLoad(true);
       console.log("error", e);
     }
-  }
+  };
 
   useEffect(() => {
     if (!taps.length && !initialLoad) {
       getInitialTaps();
     }
   }, [taps, setInitialLoad, initialLoad, setTaps]);
 
+  const LANG_PREF_KEY = "userLanguage";
+
+  // Initial page load: get browser's default language and init localization
+  useEffect(() => {
+    const language = window.navigator.userLanguage || window.navigator.language;
+    const supportedLanguages = ["en", "ja"];
+
+    // Check if 'en' or 'ja' sub-strings are in the default's language: should handle
+    // particular cases such as en-GB, en-US, etc.
+    if (language.includes("en")){
```

Comment:
Yes, I agree that `switch` is better than a series of `else if`.

In this code, I thought the `else if` block is redundant can can be removed since `ja` is the default. I could be wrong though.

Reply:
Tagging @juliwithoutthee for the FunFact feature. The functionality of the Fun Fact component is complete. It just needs to be added to the page and styled.